File: AndroidRCLGroupName/app/src/main/java/com/example/shakil/androidrclgroupname/Adapter/AlphabetAdapter.java
package com.example.shakil.androidrclgroupname.Adapter;

import android.graphics.Color;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.amulyakhare.textdrawable.TextDrawable;
import com.example.shakil.androidrclgroupname.Common.Common;
import com.example.shakil.androidrclgroupname.Interface.IOnAlphabetClickListener;
import com.example.shakil.androidrclgroupname.R;

import java.util.List;

public class AlphabetAdapter extends RecyclerView.Adapter<AlphabetAdapter.MyViewHolder> {

    List<String> alphabetList;
    IOnAlphabetClickListener iOnAlphabetClickListener;

    public void setiOnAlphabetClickListener(IOnAlphabetClickListener iOnAlphabetClickListener) {
        this.iOnAlphabetClickListener = iOnAlphabetClickListener;
    }

    public AlphabetAdapter() {
        alphabetList = Common.getAlphabet();
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View itemView = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.alphabet_item, viewGroup, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, final int i) {
        TextDrawable drawable;
        //Get available position of character
        final int available_position = Common.alphabet_available.indexOf(alphabetList.get(i));

        //If character is available in list
        if (available_position != -1) {
            drawable = TextDrawable.builder().buildRound(alphabetList.get(i), Color.GREEN);
        } else {
            //If character is not in alphabet , just set GRAY
            drawable = TextDrawable.builder().buildRound(alphabetList.get(i), Color.GRAY);
        }

        myViewHolder.alphabet_avatar.setImageDrawable(drawable);
        myViewHolder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // When user choose alphabet character, just go back MainActivity and scroll to position of first character
                iOnAlphabetClickListener.onAlphabetClickListener(alphabetList.get(i), i);
            }
        });
    }

    @Override
    public int getItemCount() {
        return alphabetList.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        ImageView alphabet_avatar;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            alphabet_avatar = itemView.findViewById(R.id.alphabet_avatar);
        }
    }
}

File: KotlinRCLGroupName/app/src/main/java/com/example/shakil/kotlinrclgroupname/Adapter/AlphabetAdapter.kt
package com.example.shakil.kotlinrclgroupname.Adapter

import android.graphics.Color
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.amulyakhare.textdrawable.TextDrawable
import com.example.shakil.kotlinrclgroupname.Common.Common
import com.example.shakil.kotlinrclgroupname.Interface.IOnAlphabetItemClickListener
import com.example.shakil.kotlinrclgroupname.R

class AlphabetAdapter : RecyclerView.Adapter<AlphabetAdapter.MyViewHolder>() {

    internal var alphabetList: List<String>
    internal lateinit var iOnAlphabetItemClick: IOnAlphabetItemClickListener

    init {
        alphabetList = Common.genAlphabetList()
    }

    fun setAlphabetClick(iOnAlphabetClickListener: IOnAlphabetItemClickListener) {
        iOnAlphabetItemClick = iOnAlphabetClickListener
    }

    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): AlphabetAdapter.MyViewHolder {
        val itemView = LayoutInflater.from(p0.context).inflate(R.layout.alphabet_item, p0, false)
        return MyViewHolder(itemView)
    }

    override fun getItemCount(): Int {
        return alphabetList.size
    }

    override fun onBindViewHolder(p0: AlphabetAdapter.MyViewHolder, p1: Int) {
        val drawable: TextDrawable
        //Get available position of character
        val available_position = Common.alphabet_available.indexOf(alphabetList[p1])

        //If character is available in list
        if (available_position != -1) {
            drawable = TextDrawable.builder().buildRound(alphabetList[p1], Color.GREEN)
        } else {
            //If character is not in alphabet , just set GRAY
            drawable = TextDrawable.builder().buildRound(alphabetList[p1], Color.GRAY)
        }

        p0.alphabet_item.setImageDrawable(drawable)
        p0.itemView.setOnClickListener {
            // When user choose alphabet character, just go back MainActivity and scroll to position of first character
            iOnAlphabetItemClick.onAlphabetItemClick(alphabetList[p1], available_position)
        }
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var alphabet_item: ImageView
        init {
            alphabet_item = itemView.findViewById(R.id.alphabet_item) as ImageView
        }
    }
}